src/chan_vese.jl

```diff
@@ -0,0 +1,252 @@
+using MosaicViews
+using LazyArrays
+using BenchmarkTools
+using Images, TestImages
+using ImageBase.ImageCore: GenericGrayImage, GenericImage
+
+function calculate_averages(img::AbstractArray{T, N}, H𝚽::AbstractArray{T, M}) where {T<:Real, N, M}
+    H𝚽ⁱ = @. 1. - H𝚽
+    ∫H𝚽 = sum(H𝚽)
+    ∫H𝚽ⁱ = sum(H𝚽ⁱ)
+    if ndims(img) == 2
+        ∫u₀H𝚽 = sum(img .* H𝚽)
+        ∫u₀H𝚽ⁱ = sum(img .* H𝚽ⁱ)
+    elseif ndims(img) == 3
+        ∫u₀H𝚽 = sum(img .* H𝚽, dims=(1, 2))
+        ∫u₀H𝚽ⁱ = sum(img .* H𝚽ⁱ, dims=(1, 2))
+    end
+    if ∫H𝚽 != 0
+        c₁ = ∫u₀H𝚽 / ∫H𝚽
+    end
+    if ∫H𝚽ⁱ != 0
+        c₂ = ∫u₀H𝚽ⁱ / ∫H𝚽ⁱ
+    end
+
+    return c₁, c₂
+end
+
+function difference_from_average_term(img::AbstractArray{T, N}, H𝚽::AbstractArray{T, M}, λ₁::Float64, λ₂::Float64) where {T<:Real, N, M}
+    c₁, c₂ = calculate_averages(img, H𝚽)
+
+    if ndims(img) == 2
+        return @. -λ₁ * (img - c₁)^2 + λ₂ * (img - c₂)^2
```

Comment:
If you make `H𝚽` have the same dimensionality as `img` then this line pertains to the whole thing.